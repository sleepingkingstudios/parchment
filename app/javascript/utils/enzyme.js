import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
