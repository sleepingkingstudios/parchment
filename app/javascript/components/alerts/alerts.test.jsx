import React from 'react';
import { shallow } from 'enzyme';

import Alerts from './alerts';

describe('<Alerts />', () => {
  const alertsData = [
    {
      id: '00000000-0000-0000-0000-000000000000',
      message: 'Alert with Message',
    },
    {
      id: '00000000-0000-0000-0000-000000000001',
      alertStyle: 'danger',
      message: 'Alert with Style',
    },
    {
      id: '00000000-0000-0000-0000-000000000002',
      children: (
        <strong>Alert with Children</strong>
      ),
    },
  ];
  const defaultProps = { alerts: alertsData };

  it('should render each alert', () => {
    const rendered = shallow(<Alerts {...defaultProps} />);
    const alerts = rendered.find('Alert');

    expect(alerts.length).toEqual(alertsData.length);

    alerts.forEach((alert, index) => {
      const data = alertsData[index];

      expect(alert).toHaveDisplayName('Alert');
      expect(alert).toHaveProp('alert', data);
    });
  });

  it('should match the snapshot', () => {
    const rendered = shallow(<Alerts {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });
});
