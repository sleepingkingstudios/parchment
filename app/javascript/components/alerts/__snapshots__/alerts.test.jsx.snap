// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerts /> should match the snapshot 1`] = `
Array [
  <Alert
    alert={
      Object {
        "id": "00000000-0000-0000-0000-000000000000",
        "message": "Alert with Message",
      }
    }
    className={null}
    dismissAlert={[MockFunction]}
    key="00000000-0000-0000-0000-000000000000"
  />,
  <Alert
    alert={
      Object {
        "alertStyle": "danger",
        "id": "00000000-0000-0000-0000-000000000001",
        "message": "Alert with Style",
      }
    }
    className={null}
    dismissAlert={[MockFunction]}
    key="00000000-0000-0000-0000-000000000001"
  />,
  <Alert
    alert={
      Object {
        "children": <strong>
          Alert with Children
        </strong>,
        "id": "00000000-0000-0000-0000-000000000002",
      }
    }
    className={null}
    dismissAlert={[MockFunction]}
    key="00000000-0000-0000-0000-000000000002"
  />,
]
`;
